opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1453,66 +1497,75 @@ void CpGrid::addLgrUpdateLeafView(const std::array<int,3>& cells_per_dim, const
                 break;
         }
-        if(!is_there_corn) { // corner is not involved in refinement, so we store it.
+        if(!is_there_allPatchCorn) { // corner is not involved in any refinement, so we store it.
             level_to_leaf_corners[{0, corner}] = corner_count;
             corner_count +=1;
         }
-    }
-    // Corners coming from level1, i.e. refined (new-born) corners.
-    for (int corner = 0; corner < this -> data_[1]->size(3); ++corner) {
-        level_to_leaf_corners[{1, corner}] = corner_count;
-        corner_count +=1;
+    } // end corner-forloop
+    // Corners coming from LGRs, i.e. refined (new-born) corners.
+    for (int patch = 0; patch < num_patches; ++patch){
+        for (int corner = 0; corner < this -> data_[patch+1]->size(3); ++corner) {
+            level_to_leaf_corners[{patch + 1, corner}] = corner_count;
+            corner_count +=1;
+        }
     }
     // Resize the container of the leaf view corners.
     leaf_corners.resize(corner_count);
-    for (const auto& [level_cornIdx, leafCornIdx] : level_to_leaf_corners) { // level_cornIdx = {level, corner index}
+    for (const auto& [level_cornIdx, leafCornIdx] : level_to_leaf_corners) { // level_cornIdx[0] = level, level_cornIdx[1] = corner index
         const auto& level_data = *(this->data_[level_cornIdx[0]]);
         leaf_corners[leafCornIdx] = level_data.geometry_.geomVector(std::integral_constant<int,3>()).get(level_cornIdx[1]);
     }
-    // Map to relate boundary patch corners with their equivalent refined/new-born ones. {0,oldCornerIdx} -> {1,newCornerIdx}
+    // Map to relate boundary patches corners with their equivalent refined/new-born ones. {0,oldCornerIdx} -> {level,newCornerIdx}
     std::map<std::array<int,2>, std::array<int,2>> old_to_new_boundaryPatchCorners;
     // To store (indices of) boundary patch corners.
-    std::vector<int> boundary_patch_corners;
-    boundary_patch_corners.reserve(boundary_old_to_new_corners.size());
-    for (long unsigned int corner = 0; corner < boundary_old_to_new_corners.size(); ++corner) {
-        old_to_new_boundaryPatchCorners[{0, boundary_old_to_new_corners[corner][0]}] = {1, boundary_old_to_new_corners[corner][1]};
-        boundary_patch_corners.push_back(boundary_old_to_new_corners[corner][0]);
+    std::vector<std::vector<int>> all_boundary_corners; // CHANGE NAME
+    all_boundary_corners.resize(num_patches);
+    for (int patch = 0; patch < num_patches; ++patch){
+        all_boundary_corners[patch].reserve(all_patch_boundary_old_to_new_corners[patch].size());
+        for (int corner = 0; corner < static_cast<int>(all_patch_boundary_old_to_new_corners[patch].size()); ++corner) {
+            old_to_new_boundaryPatchCorners[{0, all_patch_boundary_old_to_new_corners[patch][corner][0]}]
+                = {patch +1, all_patch_boundary_old_to_new_corners[patch][corner][1]};
+            // shifted +1 since coarse grid is level 0, levels/LGRs are 1,2, ..., num_patches. 
```

Comment:
Comment does not fit next line.

Reply:
the comment refers to "patch +1" and not to  "..[patch]". I can modify it, in any case.